client/my-sites/domains/domain-management/edit/domain-types/mapped-domain-type.jsx

```diff
@@ -0,0 +1,284 @@
+		);
+	}
+
+	domainConnectMappingNavItem() {
+		const { supportsDomainConnect, hasWpcomNameservers, pointsToWpcom } = this.props.domain;
+		if ( ! supportsDomainConnect || hasWpcomNameservers || pointsToWpcom ) {
+			return;
+		}
+
+		const path = domainManagementDomainConnectMapping(
+			this.props.selectedSite.slug,
+			this.props.domain.name
+		);
+
+		return (
+			<VerticalNavItem path={ path }>
+				{ this.props.translate( 'Connect Your Domain' ) }
+			</VerticalNavItem>
+		);
+	}
+
+	transferMappedDomainNavItem() {
+		const { domain, selectedSite, translate } = this.props;
+
+		if ( domain.expired || domain.isSubdomain || ! domain.isEligibleForInboundTransfer ) {
+			return null;
+		}
+
+		const path = domainTransferIn( selectedSite.slug, domain.name, true );
+
+		return (
+			<VerticalNavItem path={ path }>
+				{ translate( 'Transfer Domain to WordPress.com' ) }
+			</VerticalNavItem>
+		);
+	}
+
+	resolveStatus() {
+		const { domain, translate, moment } = this.props;
+		const { expiry } = domain;
+
+		if ( isExpiringSoon( domain, 30 ) ) {
+			const expiresMessage = translate( 'Expires in %(days)s', {
+				args: { days: moment.utc( expiry ).fromNow( true ) },
+			} );
+
+			if ( isExpiringSoon( domain, 5 ) ) {
+				return {
+					statusText: expiresMessage,
+					statusClass: 'status-error',
+					icon: 'info',
+				};
+			}
+
+			return {
+				statusText: expiresMessage,
+				statusClass: 'status-warning',
+				icon: 'info',
+			};
+		}
+
+		if ( ! domain.pointsToWpcom ) {
+			return {
+				statusText: translate( 'Action required' ),
+				statusClass: 'status-error',
+				icon: 'info',
+			};
+		}
+
+		return {
+			statusText: translate( 'Active' ),
+			statusClass: 'status-success',
+			icon: 'check_circle',
+		};
+	}
+
+	renderExpiringSoon() {
+		const { domain, translate, moment } = this.props;
+		const { expiry } = domain;
+
+		if ( ! isExpiringSoon( domain, 30 ) ) {
+			return (
+				<div>
+					{ translate(
+						'Your domain will expire in {{strong}}%(days)s{{/strong}}. Please renew it before it expires or it will stop working.',
+						{
+							components: {
+								strong: <strong />,
+							},
+							args: {
+								days: moment.utc( expiry ).fromNow( true ),
+							},
+						}
+					) }
+				</div>
+			);
+		}
+
+		return null;
+	}
+
+	renderSettingUpNameservers() {
+		const { domain, selectedSite, translate } = this.props;
+		if (
+			selectedSite.jetpack ||
+			( selectedSite.options && selectedSite.options.is_automated_transfer )
+		) {
+			return null;
+		}
+
+		if ( domain.pointsToWpcom ) {
+			return null;
+		}
+
+		let primaryMessage;
+		let learnMoreLink;
+
+		if ( isSubdomain( domain.name ) ) {
+			learnMoreLink = <a href={ MAP_SUBDOMAIN } target="_blank" rel="noopener noreferrer" />;
+			primaryMessage = translate(
+				'Your domain mapping has not been setup. You need to create the relevant CNAME or NS records at your current DNS provider. {{learnMoreLink}}Learn more{{/learnMoreLink}}',
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?